packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2056,6 +2124,46 @@ Hardhat Network's forking functionality only works with blocks from at least spu
   public isEip1559Active(): boolean {
     return this._vm._common.gteHardfork("london");
   }
+
+  private async _getEstimateGasFeePriceFields(
+    callParams: CallParams,
+    blockNumberOrPending: BN | "pending"
+  ): Promise<
+    { gasPrice: BN } | { maxFeePerGas: BN; maxPriorityFeePerGas: BN }
+  > {
+    if (!this.isEip1559Active() || callParams.gasPrice !== undefined) {
+      return { gasPrice: callParams.gasPrice ?? (await this.getGasPrice()) };
+    }
+
+    let maxFeePerGas = callParams.maxFeePerGas;
+    let maxPriorityFeePerGas = callParams.maxPriorityFeePerGas;
+
+    if (maxPriorityFeePerGas === undefined) {
+      maxPriorityFeePerGas = await this.getMaxPriorityFeePerGas();
+
+      if (
+        callParams.maxFeePerGas !== undefined &&
+        callParams.maxFeePerGas.lt(maxPriorityFeePerGas)
+      ) {
+        maxPriorityFeePerGas = callParams.maxFeePerGas;
+      }
```

Comment:
I would move this after `maxFeePerGas`, where it will be defined:

```ts
if (maxFeePerGas.lt(maxPriorityFeePerGas)) {
  maxPriorityFeePerGas = maxFeePerGas;
}
```

Not just because there's one check less, but because it makes more conceptual sense to me. Plus, right now the `maxFeePerGas` set in the following block is always greater or equal than `maxPriorityFeePerGas`, but that might be just a circumstantial thing.

That being said, maybe there is a good reason why this _needs_ to be done here?

Reply:
The default `maxFeePerGas` is defined based on `maxPriorityFeePerGas`, so it's not as simple as moving it.

We could do an `else` for `if (maxFeePerGas === undefined) {`, but that would look even worse imo.

Reply:
Oh, that makes sense, disregard this then!